cmd/notary/keys.go

```diff
@@ -338,7 +365,56 @@ func (k *keyCommander) keysImportRoot(cmd *cobra.Command, args []string) error {
 	}
 	defer importFile.Close()
 
-	err = cs.ImportRootKey(importFile)
+	pemBytes, err := ioutil.ReadAll(importFile)
+	if err != nil {
+		return fmt.Errorf("Error reading input file: %v", err)
+	}
+
+	pemRole := trustmanager.ReadRoleFromPEM(pemBytes)
+
+	// Rewind after reading the first time
+	_, err = importFile.Seek(0, 0)
+	if err != nil {
+		return fmt.Errorf("Error reading input file: %v", err)
+	}
+
+	if pemRole != "" && !data.ValidRole(pemRole) {
```

Comment:
Technically for delegation keys we don't actually check the PEM header roles when signing, so it'd be fine to have an invalid name.  I'm fine changing this, I like the flexibility of being able to use a notary command or just dropping keys in and not having to interface directly with the `tuf_keys` directory.